packages/angular/ssr/test/app-engine_spec.ts

```diff
@@ -0,0 +1,136 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.dev/license
+ */
+
+/* eslint-disable import/no-unassigned-import */
+import 'zone.js/node';
+import '@angular/compiler';
+/* eslint-enable import/no-unassigned-import */
+
+import { AngularServerApp } from '../src/app';
+import { AngularAppEngine } from '../src/app-engine';
+import { setAngularAppEngineManifest } from '../src/manifest';
+import { setAngularAppTestingManifest } from './testing-utils';
+
+describe('AngularAppEngine', () => {
+  let appEngine: AngularAppEngine;
+  // Enable dev mode for testing purposes.
+  // @ts-expect-error Property 'isDevMode' does not exist on type 'typeof AngularAppEngine'.
+  AngularAppEngine.isDevMode = true;
+
+  describe('Localized app', () => {
+    const LOCALES = ['it', 'en'];
+
+    beforeAll(() => {
+      setAngularAppEngineManifest({
+        entryPoints: new Map(
+          LOCALES.map((locale) => [
+            locale,
+            async () => {
+              setAngularAppTestingManifest(locale);
+
+              return { AngularServerApp };
+            },
+          ]),
+        ),
+        basePath: '',
+      });
+
+      appEngine = new AngularAppEngine();
+    });
+
+    it('should return null for requests to unknown pages', async () => {
+      const request = new Request('https://example.com/unknown/page');
+      const response = await appEngine.render(request);
+      expect(response).toBeNull();
+    });
+
+    it('should return null for requests with unknown locales', async () => {
+      const request = new Request('https://example.com/es/home');
+      const response = await appEngine.render(request);
+      expect(response).toBeNull();
+    });
+
+    for (const locale of LOCALES) {
```

Comment:
Suggestion: Do `en` and `it` use different code paths in this scenario? If not, I'd just omit this `for` loop altogether for simplicity and use `it` (the non-"default" option) for all the tests.